go/interpreter/lfvm/gas_test.go

```diff
@@ -59,3 +61,99 @@ func TestGas_CallGasCalculation(t *testing.T) {
 		})
 	}
 }
+
+func TestGas_AddressInAccessList(t *testing.T) {
+
+	tests := map[string]struct {
+		setup        func(*context)
+		accessStatus bool
+		coldCost     tosca.Gas
+	}{
+		"istanbul": {
+			setup: func(c *context) {
+				c.params.Revision = tosca.R07_Istanbul
+			},
+			accessStatus: true,
+		},
+		"berlin_warm": {
+			setup: func(c *context) {
+				c.params.Revision = tosca.R09_Berlin
+				c.context.(*tosca.MockRunContext).EXPECT().IsAddressInAccessList(gomock.Any()).Return(true)
+				c.stack.push(uint256.NewInt(1))
+				c.stack.push(uint256.NewInt(2))
+			},
+			accessStatus: true,
+			coldCost:     ColdAccountAccessCostEIP2929 - WarmStorageReadCostEIP2929,
+		},
+		"berlin_cold_enough_gas": {
+			setup: func(c *context) {
+				c.params.Revision = tosca.R09_Berlin
+				c.context.(*tosca.MockRunContext).EXPECT().IsAddressInAccessList(gomock.Any()).Return(false)
+				c.context.(*tosca.MockRunContext).EXPECT().AccessAccount(gomock.Any()).Return(tosca.ColdAccess)
+				c.stack.push(uint256.NewInt(1))
+				c.stack.push(uint256.NewInt(2))
+				c.gas = ColdAccountAccessCostEIP2929 - WarmStorageReadCostEIP2929
+			},
+			accessStatus: false,
+			coldCost:     ColdAccountAccessCostEIP2929 - WarmStorageReadCostEIP2929,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			runContext := tosca.NewMockRunContext(ctrl)
+
+			ctxt := context{
+				status:  statusRunning,
+				params:  tosca.Parameters{},
+				context: runContext,
+				stack:   NewStack(),
+				memory:  NewMemory(),
+				gas:     1 << 20,
+			}
+			test.setup(&ctxt)
+
+			warmGot, coldCostGot, errGot := addressInAccessList(&ctxt)
+
+			if errGot != nil {
+				t.Errorf("unexpected error, wanted %v, got %v", nil, errGot)
+			}
+
+			if warmGot != test.accessStatus {
+				t.Errorf("unexpected warm access status, wanted %v, got %v", test.accessStatus, warmGot)
+			}
+
+			if coldCostGot != test.coldCost {
+				t.Errorf("unexpected cold cost, wanted %d, got %d", test.coldCost, coldCostGot)
+			}
+
+		})
+	}
+}
+
+func TestGas_AddressInAccessListReportsOutOfGas(t *testing.T) {
+
+	ctrl := gomock.NewController(t)
+	runContext := tosca.NewMockRunContext(ctrl)
+	ctxt := context{
+		status:  statusRunning,
+		params:  tosca.Parameters{},
+		context: runContext,
+		stack:   NewStack(),
+		memory:  NewMemory(),
+		gas:     1 << 20,
+	}
+	ctxt.params.Revision = tosca.R09_Berlin
```

Comment:
nit: can we initialize data members in line 139?  And leave here manipulations (like stack)